catalog/app/utils/CatalogSettings.tsx

```diff
@@ -0,0 +1,78 @@
+import type { S3 } from 'aws-sdk'
+import * as React from 'react'
+
+import * as AWS from 'utils/AWS'
+import * as Config from 'utils/Config'
+import * as Cache from 'utils/ResourceCache'
+
+const CONFIG_KEY = 'catalog/settings.json'
+
+export interface CatalogSettings {
+  customNavLink?: {
+    url: string
+    label: string
+  }
+}
+
+async function fetchSettings({
+  s3,
+  serviceBucket,
+  mode,
+}: {
+  s3: S3
+  serviceBucket: string
+  mode: string
+}) {
+  if (mode === 'MARKETING') return null
+
+  const location = `s3://${serviceBucket}/${CONFIG_KEY}`
+  try {
+    const res = await s3.getObject({ Bucket: serviceBucket, Key: CONFIG_KEY }).promise()
+    const text = res.Body!.toString('utf-8')
+    return JSON.parse(text) as CatalogSettings
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.warn(`Error fetching catalog settings from "${location}":`)
+    // eslint-disable-next-line no-console
+    console.error(e)
+    return null
+  }
+}
+
+const CatalogSettingsResource = Cache.createResource({
+  name: 'CatalogSettings.config',
+  fetch: fetchSettings,
+  // @ts-expect-error
+  key: () => null,
```

Comment:
What is a meaning of `key === null`? Don't save to cache?

Probably, instead of explaining, it's better to have `Cache.createResourceOnSomeAdditionalTerms` method